src/components/CoboVault/CoboVaultRawReader.jsx

```diff
@@ -0,0 +1,249 @@
+import { Button, Grid, Modal } from "@material-ui/core";
+import QrReader from "react-qr-reader";
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import URProgress from "./URProgress";
+
+const QR_CODE_READER_DELAY = 300; // ms?
+
+const QRScanner = (props) => {
```

Comment:
Can the scanner and reader be in their own files too? It might be good to have a dedicated `QR` directory. These are the components I've seen so far that could go there: `QRScanner`, `QRReader`, `QRPlayer`